opal/mca/common/verbs/common_verbs_basics.c

```diff
@@ -61,3 +64,30 @@ bool opal_common_verbs_check_basics(void)
     return true;
 }
 
+int opal_common_verbs_fork_test(void)
+{
+    /* Make sure that ibv_fork_init is called before the calls to other memory registering verbs,
+     * which will be called after this function */
+#ifdef HAVE_IBV_FORK_INIT
+    if (0 != opal_verbs_want_fork_support) {
+        /* Check if fork support is requested by the user */
+        if (0 != ibv_fork_init()) {
+            opal_show_help("help-opal-common-verbs.txt", "ibv_fork_init fail",
```

Comment:
The rationale here is that most users don't want/care about fork support.  So we need a way for them to *not* get warnings about fork support if it doesn't work.

Hence, the default value (negative) is intended to be "try for fork support, but if it doesn't work, just silently keep going."

For those who *do* want fork() support, they can explicitly ask for it by setting the MCA var to a positive value.  This is not only an indication that we definitely need ibv_fork_init() to succeed, the user also needs to know if it fails.  I.e., it's "try for fork support, and if it doesn't work, print an error and kill the job."

Make sense?